tools/metricq-summary/src/summary.cpp

```diff
@@ -28,13 +28,23 @@
 
 #include "summary.hpp"
 
-Summary Summary::calculate(std::vector<metricq::TimeValue>&& tv_pairs)
+Summary Summary::calculate(std::vector<metricq::TimeValue>&& tv_pairs, std::chrono::milliseconds start_delta, std::chrono::milliseconds stop_delta)
 {
-    assert(tv_pairs.size() > 0);
-
     auto begin = tv_pairs.begin();
     auto end = tv_pairs.end();
 
+    auto start = tv_pairs.front().time;
```

Comment:
These two lines only work, if you get a response with at least one timestamp value pair.
aka `assert(tv_pairs.size() > 0);`